jtuples/src/main/java/org/jtuples/Functional.java

```diff
@@ -0,0 +1,210 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jtuples;
+
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * This class provides static methods to support a functional programming style.
+ * It provides methods commonly seen in functional programming environments,
+ * that revolve around the use of tuples and functions (or lambdas).
+ * @author Andre Santos <contact.andre.santos@gmail.com>
+ */
+public final class Functional {
+    private Functional() {
+        throw new AssertionError();
+    }
+
+    /**
+     * Returns a curried version of the given function.
+     * Given a function that accepts a pair as its argument, its curried
+     * version is a function that accepts the two components of the pair
+     * individually.
+     * @param <A> the type of the first component of the pair
+     * @param <B> the type of the second component of the pair
+     * @param <C> the return type
+     * @param function the function to curry
+     * @return the curried version of the given function
+     */
+    public static <A, B, C> BiFunction<A, B, C> curry(
+            Function<Pair<A, B>, C> function) {
+        return (A a, B b) -> function.apply(new Pair<>(a, b));
+    }
+
+    /**
+     * Returns an uncurried version of the given function.
+     * Given a function that accepts two individual parameters, its uncurried
+     * version accepts instead a pair consisting of the two parameters.
+     * @param <A> the type of the first parameter
+     * @param <B> the type of the second parameter
+     * @param <C> the return type
+     * @param function the function to uncurry
+     * @return the uncurried version of the given function
+     */
+    public static <A, B, C> Function<Pair<A, B>, C> uncurry(
+            BiFunction<A, B, C> function) {
+        return (Pair<A, B> p) -> function.apply(p.first(), p.second());
+    }
+
+    /**
+     * Returns the split of the two given functions.
+     * Given two functions with parameters of the same type, the split of
+     * such functions is a single function, with the same parameters, that
+     * returns a pair whose components are the application of the original
+     * functions to the arguments.
```

Comment:
This is good, but to make it more clear could you also add a "more visual" explanation to this, or maybe an example? Something like A->B and A->C producing A->(B, C).